test/preview.test.ts

```diff
@@ -77,7 +77,8 @@ describe('A Preview plugin', () => {
             window.removeEventListener(event, listener);
         }
 
-        jest.resetAllMocks();
+        jest.restoreAllMocks();
+        jest.clearAllMocks();
```

Comment:
I choose to use restore over reset because the reset replaces the mock with an empty function that return undefined, leading some tests to errors since we are mocking the `window.removeEventListener` and `window.addEventListener`, on the other hand, the restore will restore the original function, and since we are spying the window again in the `beforeEach`, I see no problem.

The `clearAllMocks` is here because the `restoreMock` restore only `jest.spyOn` mocks, so the clear will deal with `jest.fn()` mocks

Reply:
[Relevant docs](https://jestjs.io/docs/jest-object#jestclearallmocks).

`clearAllMocks` clears the call history for every mock. Good.

`resetAllMocks` calls `clearAllMocks` plus replaces the mock implementation with a function returning undefined. If that is the problematic behavior, we should replace the call with _just_ `clearAllMocks`.

`restoreAllMocks` _removes_ the mock. After this call, all mocks that can be restored are _no longer mocked_. This might open up potential flaky tests, regressions, and false negatives since we no longer mock the situation we want to test. If there is a test after this call validating that internal errors are not propagated, for example, it would become a non-error scenario, and the code would no longer be tested.